File: python/1.two_sum.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen = {}

        for index, num in enumerate(nums):
            diff = target - num
            if diff in seen:
                result = [seen[diff], index]
                return result
            else:
                seen[num] = index

File: cpp/1.two_sum.cpp
#include <unordered_map>
#include <vector>

class Solution {
public:
    std::vector<int> twoSum(std::vector<int>& nums, int target) {
        std::unordered_map<int, int> seen;
        std::vector<int> result;

        unsigned i = 0;

        for (auto& num : nums) {
            int diff = target - num;
            if (seen.contains(diff)) {
                result.push_back(seen[diff]);
                result.push_back(i);
                return result;
            }
            else {
                seen[num] = i;
                ++i;
            }
        }
        return {};
    }
};
